water-gui.py

```diff
@@ -238,11 +242,19 @@ def evaluate(self) -> None:
         # done? redirect to self.update_db()
         if messagebox.askyesno("¿Seguir?", """¿Desea seguir con el proceso usando las variables definidas?
 (Este proceso no se puede deshacer)"""):
+            if messagebox.askyesno("¿Registrar cobro?", """¿Desea registrar un ingreso al negocio por el producto tomado?
+
+Si no, se le va a redirigir a una pagina para
+que explique sus motivos."""):
+                reason = self.no_entry(product, vendor, unit_count)
+            else:
+                reason = "N/A"

```

Comment:
No matter if the vendor registered a monetary entry or not, we'll always generate a `reason` on the database.